code/src/reducers/tasks.js

```diff
@@ -0,0 +1,57 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const initialState = {
+  taskData: [],
+  total: 0
+};
+
+export const tasks = createSlice({
+  name: "tasks",
+  initialState,
+  reducers: {
+    addTask: (state, action) => {
+      let { text, id } = action.payload;
+
+      state.taskData.push({
+        id,
+        text,
+        complete: false,
+        important: false
+      });
+
+      state.total += 1;
+    },
+    toggle: (state, action) => {
+      const todo = state.taskData.find(todo => todo.id === action.payload);
+
+      if (todo) {
+        todo.complete = !todo.complete;
+      }
+    },
+    toggleImportant: (state, action) => {
+      const todo = state.taskData.find(todo => todo.id === action.payload);
+
+      if (todo) {
+        todo.important = !todo.important;
+      }
+    },
+    deleteAll: state => {
+      state.taskData = [];
+      state.total = 0;
+    },
+    deleteOne: (state, action) => {
+      state.taskData = state.taskData.filter(
+        item => item.id !== action.payload
+      );
+      state.total -= 1;
+    },
+    showCompleted: state => {
+      state.taskData = state.taskData.filter(todo => todo.complete);
+      state.total = state.taskData.length;
+    },
+    showNotCompleted: state => {
+      state.taskData = state.taskData.filter(todo => !todo.complete);
+      state.total = state.taskData.length;
+    }

```

Comment:
This doesn't really do what you want it to do I think.
When you have a list of 5 todos, for example. And then 2 of them are completed and 3 of them are not.
If you click to show only the completed ones, and then the ongoing ones you'll get no ongoing ones left. Since you made the taskData now be only the completed ones. This also messes up the total todos count.
You should modify the global state when you just want to filter the todos shown in different categories.
Does this make sense? If not, let me know!

Reply:
I think I know what you mean. I was sure it does what it suppose to, but now I know it is not. I will check this out 😆